packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -73,18 +79,31 @@ export class BlockToolAdapter {
    * @param input - input element
    * @param key - data key input is attached to
    */
-  #handleBeforeInputEvent(event: InputEvent, input: HTMLElement, key: DataKey): void {
+  #handleBeforeInputEvent = (event: InputEvent, input: HTMLElement, key: DataKey): void => {
     /**
      * We prevent all events to handle them manually via model update
      */
     event.preventDefault();
 
+    const nativeInput = this.#mode === InputMode.Native;
```

Comment:
it is the main problem right now, BlockToolAdapter should support several inputs